packages/notifier/src/asyncIterableAdaptor.js

```diff
@@ -80,6 +80,8 @@ export const makeAsyncIterableFromNotifier = notifierP => {
                 return harden({ value, done });
               });
           }
+          // xxx hint for type checker
+          assert(myIterationResultP);
```

Comment:
Ugh that's weird. Why doesn't flow control realize that `myIterationResultP` is never `undefined` here?

Reply:
Yeah, it's weird. I don't now why but here's the error without it,
```
> @agoric/ui-components@0.3.3 lint:types
> tsc -p jsconfig.json

../notifier/src/asyncIterableAdaptor.js(39,3): error TS2322: Type '{ [Symbol.asyncIterator]: () => { next: () => Promise<{ value: T; done: boolean; }> | undefined; } & RemotableBrand<{}, { next: () => Promise<{ value: T; done: boolean; }> | undefined; }>; } & RemotableBrand<...>' is not assignable to type 'ConsistentAsyncIterable<T>'.
  The types returned by '[Symbol.asyncIterator]().next(...)' are incompatible between these types.
    Type 'Promise<{ value: T; done: boolean; }> | undefined' is not assignable to type 'Promise<IteratorResult<T, T>>'.
      Type 'undefined' is not assignable to type 'Promise<IteratorResult<T, T>>'.
```

Reply:
Oh I think I know. TS doesn't know that the chained calls, and in particular `.then()` doesn't execute the callbacks synchronously, so it can't know that `myIterationResultP = undefined;` hasn't happened.